packages/cli/src/commands/inspect/index.ts

```diff
@@ -92,20 +99,46 @@ export default async function inspect(client: Client) {
   );
 
   const until = Date.now() + timeout;
-  const wait = argv['--wait'];
+  const wait = parsedArguments.flags['--wait'] ?? false;
+  const withLogs = parsedArguments.flags['--logs'];
 
   // resolve the deployment, since we might have been given an alias
   let deployment = await getDeployment(client, contextName, deploymentIdOrHost);
 
+  let abortController: AbortController | undefined;
+  if (withLogs) {
+    if (wait) {
+      abortController = displayBuildLogs(client, deployment, true);
+    } else {
+      await displayBuildLogs(client, deployment, false);
+      return exitCode(deployment.readyState, print);
+    }
+  }
+
   while (Date.now() < until) {
-    if (!wait || !isDeploying(deployment.readyState)) {
+    if (!wait) {
       break;
     }
 
     await sleep(250);
 
     // check the deployment state again
     deployment = await getDeployment(client, contextName, deploymentIdOrHost);
+    if (!isDeploying(deployment.readyState)) {
+      abortController?.abort();
+      if (!isReady(deployment) && !isFailed(deployment)) {
+        print(
+          chalk.bold(
+            `\n\nstop waiting for logs after ${timeout}s. Deployment is ${deployment.readyState}.\n`
+          )
+        );
+      }
```

Comment:
I'm not sure I understand the logic here. The message says that it stops waiting after a timeout, but I can't see that in the condition 🤔

Reply:
Here is the full extract
```ts
while (Date.now() < until) {
   if (!wait) {
     break;
   }

   // ...
   if (!isDeploying(deployment.readyState)) {
      abortController?.abort();
      if (!isReady(deployment) && !isFailed(deployment)) {
        print(chalk.bold(`\n\nstop waiting for logs after ${timeout}s. Deployment is ${deployment.readyState}.\n`));
      }
      if (withLogs) {
        return exitCode(deployment.readyState, print);
      } else {
        break;
      }
    }
  }

  // ... code displaying the deployment's artifacts
```

This loop only happens when waiting (while condition at the top). 
Then if the build is finished, we'll stop the waiting.
However, we don't have the same logic:
- if we were inspecting logs, then we exit the command, because we don't want to:
- if we don't inspect logs, then we print the artifacts

When the